yt/yt/server/node/exec_node/job_proxy_log_manager.cpp

```diff
@@ -0,0 +1,122 @@
+#include "job_proxy_log_manager.h"
+
+#include "bootstrap.h"
+#include "private.h"
+
+#include <yt/yt/server/lib/exec_node/config.h>
+#include <yt/yt/server/lib/scheduler/helpers.h>
+#include <yt/yt/server/node/cluster_node/config.h>
+
+#include <yt/yt/core/misc/fs.h>
+
+#include <util/datetime/base.h>
+#include <util/generic/ymath.h>
+#include <util/system/fstat.h>
+
+namespace NYT::NExecNode {
+
+////////////////////////////////////////////////////////////////////////////////
+
+TJobProxyLogManager::TJobProxyLogManager(IBootstrap* bootstrap)
+    : Bootstrap_(bootstrap)
+    , Config_(Bootstrap_->GetConfig()->ExecNode->JobProxyLogManager)
+    , Directory_(Config_->Directory)
+    , ShardingKeyLength_(Config_->ShardingKeyLength)
+    , LogsStoragePeriod_(Config_->LogsStoragePeriod)
+    , DirectoryTraversalConcurrency_(Config_->DirectoryTraversalConcurrency)
+    , AsyncSemaphore_(New<NConcurrency::TAsyncSemaphore>(Config_->DirectoryTraversalConcurrency.value_or(0)))
+{}
+
+void TJobProxyLogManager::Start()
```

Comment:
Am I right that this code is executed unconditionally? I think it is better to perfrom any actions with FS only if the new logging scheme is enabled.

Reply:
And this comment is applicable to the all methods of the class.